### ISSUE TITLE
Different Cell Embeddings When Modifying the Counts of One Cell

### ISSUE THREAD

**Issue by peterhealx** [snap-stanford/UCE#57]
Dear UCE team,

Thank you very much for your work improving masked-token pre-trained foundation models. 

I have been observing that when modifying a single cell (sample)'s count within an adata object, the embeddings of at least some other cells are also changed. For example, a dataset of 100 cells and 10,000 genes. If I set cell 2's count of gene X to 0.0 (and it was formerly > 0.0), then this not only affects cell 2's embeddings, but also some of the other cells.

To mitigate this I set --filter to False (via --no-filter modification from a previous issue), and --batch-size to 1 to prevent batching of cells and potential sharing of information. However, this has not resulted in being able to only change cell 2's embeddings in the above scenario.

Is there are othe reason for the observed behaviour for example in:

- https://github.com/snap-stanford/UCE/blob/main/evaluate.py e.g. in dataloader = DataLoader(dataset, batch_size=batch_size, ...), as I see that UCE processes cells in batches, so setting batch_size to 1 might work (but did not)
- https://github.com/snap-stanford/UCE/blob/main/data_proc/data_utils.py - e.g. during normalization of the data.

Many thanks for your help 🙏

```
#Explanatory code snippet modifying a single count in the adata file to 0.0:

import scanpy as sc
import os
import numpy as np

adata = sc.read_h5ad('data/10k_pbmcs_proc.h5ad')

#100 cell subset
adata100 = adata[0:100, :]

# Ensure gene is in var_names and cell is in obs_names
if gene in adata.var_names and cell_id in adata.obs_names:
    adata[cell_id, gene] = 0.0

adata.write('data/pbmcs_files/10k_pbmcs_proc_mod.h5ad')

# Then after saving

## Run 1
python ./eval_single_anndata.py --no-filter --batch_size 1 --adata_path /data/pbmcs_files/10k_pbmcs_proc.h5ad --protein_embeddings_dir "/data/protein_embeddings/" --nlayers 33 --model_loc /data/33l_8ep_1024t_1280.torch --dir "/data/pbmcs_files/" --species "human"

## Run 2
python ./eval_single_anndata.py --no-filter --batch_size 1 --adata_path /data/pbmcs_files/10k_pbmcs_proc_mod.h5ad --protein_embeddings_dir "/data/protein_embeddings/" --nlayers 33 --model_loc /data/33l_8ep_1024t_1280.torch --dir "/data/pbmcs_files/" --species "human"

**Comment by Yanay1** [comment]
Thanks for pointing this out, the issue is due to how False is handled when parsing the arguments for filter: https://github.com/snap-stanford/UCE/issues/53#issuecomment-2465800984
